[#58113] 14.6 Documentation: Progress reporting changes  https://comm… (#16869)
opf/openproject: docs/user-guide/time-and-costs/progress-tracking/README.md

OpenProject lets you track and monitor the progress of your work packages.

> [!IMPORTANT]
> In OpenProject 14.0, the way progress is reported and calculated has changed significantly. Refer to [this blog article](https://www.openproject.org/blog/changes-progress-work-estimates/) to follow the changes. 

> > **In OpenProject 14.6 some of the changes made in 14.0 were removed and updated**. Please read the documentation below and [this blog post](https://www.openproject.org/blog/updates-to-progress-tracking-in-14-6-based-on-user-feedback/) to understand how OpenProject handles work and progress estimates.

Reviewer: Given that the changes that made us add the first message were pretty much reverted, I wonder if we can't remove this section. It might be a bit confusing for a user reading the docs, and the history of the changes are not that relevant anymore.

Author: @psatyal sometimes the users take a while to update, so I would like to leave this information here for a bit. I can remove it with 15.0 release, hopefully by then the majority of users will have upgraded and will be familiar with the update.

- If Work is already set and you enter % Complete, Remaining Work will be automatically calculated.
- If Remaining Work is already set and you enter % Complete, Work will be automatically calculated.
- If you enter Remaining Work when Work is already set (or vice versa), % Complete will be automatically calculated.

> [!NOTE]
> If you enter a value for Remaining work that is higher than Work, you will see an error message telling you that this is not possible. You will have to enter a value lower than Work to be able to save the new value.
>Additionally, the value for Remaining work cannot be removed if a value for Work exists. If you wish to unset Remaining work, you need to also unset Work.

> [!NOTE]
> **If you enter a % Complete value of 100% when Remaining work has a value**, this will also result in an error, since Remaining work must be 0h when % Complete is 100%.

**When all values are set**

- **Increasing Work**: When you increase the value of Work, the same amount is added to Remaining Work (since the total Work has increased). This change also updates the % Complete accordingly.
- **Decreasing Work**:

>   -  If you decrease *Work*, *Remaining work* is lowered by the same amount:

Reviewer 2: ```suggestion
  -  If you decrease *Work*, *Remaining work* is lowered by the same amount.
```

>   - If you decrease *Work* by less than the current value for *Remaining work*, then *Remaining work* will be set to 0h and *% Complete* to 100%

Reviewer 2: ```suggestion
  - If you decrease *Work* by more than the current value for *Remaining work*, then *Remaining work* will be set to 0h and *% Complete* to 100%
```

- **Changing Remaining work updates % Complete**
- **Changing % Complete updates Remaining work**

> For more details and examples of progress tacking calculation please refer to this [blog article](https://www.openproject.org/blog/updates-to-progress-tracking-in-14-6-based-on-user-feedback/).

Reviewer 2: ```suggestion
For more details and examples of progress tracking calculation please refer to this [blog article](https://www.openproject.org/blog/updates-to-progress-tracking-in-14-6-based-on-user-feedback/).
```

This allows you to edit %&nbsp;Complete (by changing status) or Work and get a preview of the updated Remaining work before saving changes.

> [!NOTE]

> > In the upcoming version, statuses cannot have an empty %&nbsp;Complete value in status-based progress reporting mode.When upgrading, all statuses that do not have a value will take the default value of 0%.

Reviewer: Missing space: 

> mode.When upgrading

> > In the upcoming version, statuses cannot have an empty %&nbsp;Complete value in status-based progress reporting mode.When upgrading, all statuses that do not have a value will take the default value of 0%.

Author: ```suggestion
> In the upcoming version, statuses cannot have an empty %&nbsp;Complete value in status-based progress reporting mode. When upgrading, all statuses that do not have a value will take the default value of 0%.
```

## Hierarchy totals

OpenProject will automatically show totals for Work, Remaining work and % Complete in a work package hierarchy (any parent with children). These appear in a work package table as a number with a Σ sign next to it, indicating that it is a total of the values of the parent _and_ children.

![Hierarchy totals for Work, Remaining work and % Complete](hierarchy-totals-days.png)

OpenProject offers two modes for calculating *% Complete* in hierarchy totals:

- **Weighted by work**: The total %&nbsp;Complete value of a hierarchy is a weighted average tied to Work. For example, a feature with Work set to 50h that is 30% done will influence the total of %&nbsp;Complete of the parent more than a feature with Work set to 5h that is 70% done. 

  > [!TIP]
  >
  > Work packages without *Work* will be ignored.

> - **Simple average**: *Work* is ignored and the *total % Complete* is calculated as a simple average of the *% Complete* values from the work packages in the hierarchy.

Reviewer 2: actually, it's not exactly true: it's an average of direct children only, not an average of all descendants. And if a child has children itself, then we take its total % complete value instead of the % complete value for the computation.

```suggestion
- **Simple average**: *Work* is ignored and the *total % Complete* is calculated as a simple average of the *% Complete* values from the direct work packages children in the hierarchy. The value used on each direct child for computing the average is its *total % Complete* value, or its *% Complete* value if it has no children, or 0% if its *% Complete* value is not set.
```

  >
  > Work packages without *Work* will be ignored.

> - **Simple average**: *Work* is ignored and the *total % Complete* is calculated as a simple average of the *% Complete* values from the direct work packages children in the hierarchy. The value used on each direct child for computing the average is its *total % Complete* value, or its *% Complete* value if it has no children, or 0% if its *% Complete* value is not set.

Reviewer 2: ```suggestion
- **Simple average**: *Work* is ignored and the *total % Complete* is calculated as a simple average of the *% Complete* values from the direct work packages children in the hierarchy. The value used on each direct child for computing the average is its *total % Complete* value, or its *% Complete* value if it has no children, or 0% if its *% Complete* value is empty.
```
